Gem/Code/Source/Components/Multiplayer/PlayerJumpPadEffectComponent.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Multiplayer/Components/NetworkCharacterComponent.h>
+#include <Multiplayer/Components/NetworkTransformComponent.h>
+#include <Source/Components/Multiplayer/PlayerJumpPadEffectComponent.h>
+
+namespace MultiplayerSample
+{
+    void PlayerJumpPadEffectComponent::Reflect(AZ::ReflectContext* context)
+    {
+        AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+        if (serializeContext)
+        {
+            serializeContext->Class<PlayerJumpPadEffectComponent, PlayerJumpPadEffectComponentBase>()
+                ->Version(1);
+        }
+        PlayerJumpPadEffectComponentBase::Reflect(context);
+    }
+
+    void PlayerJumpPadEffectComponent::OnInit()
+    {
+    }
+
+    void PlayerJumpPadEffectComponent::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    void PlayerJumpPadEffectComponent::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    void PlayerJumpPadEffectComponent::ApplyJumpPadEffect(const AZ::Vector3& jumpVelocity,
+        const AZ::TimeMs& effectDuration)
+    {
+        if (HasController())
+        {
+            static_cast<PlayerJumpPadEffectComponentController*>(GetController())->ApplyJumpPadEffect(jumpVelocity, effectDuration);
+        }
+    }
+
+
+    PlayerJumpPadEffectComponentController::PlayerJumpPadEffectComponentController(PlayerJumpPadEffectComponent& parent)
+        : PlayerJumpPadEffectComponentControllerBase(parent)
+    {
+    }
+
+    void PlayerJumpPadEffectComponentController::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+        m_jumpPadEffectEvent.Enqueue(AZ::Time::ZeroTimeMs, true);
+    }
+
+    void PlayerJumpPadEffectComponentController::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+        m_jumpPadEffectEvent.RemoveFromQueue();
+    }
+
+    void PlayerJumpPadEffectComponentController::ApplyJumpPadEffect(
+        const AZ::Vector3& jumpVelocity, const AZ::TimeMs& effectDuration)
+    {
+        GetNetworkTransformComponentController()->ModifyResetCount()++;
+
+        SetIsJumpPadEffectInProgress(true);
+        SetJumpPadVector(jumpVelocity);
+        m_jumpPadEffectDuration = effectDuration;
+    }
+
+    void PlayerJumpPadEffectComponentController::ProcessInput([[maybe_unused]] Multiplayer::NetworkInput& networkInput, float deltaTime)
+    {
+        if (GetIsJumpPadEffectInProgress())
```

Comment:
This is not how I would expect a jump-pad to function. I would expect something more like a player cannon, or a trampoline. Apply an initial impulse that launches the player in a particular trajectory and then go fully ballistic on the physics immediately after the impulse is applied. Coded this way it's more like you have some kind of magic levitation buff that lifts you in a weird direction for a certain amount of time, after which when that magic buff wears off you drop back to the ground.